sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -963,7 +968,9 @@ def __init__(self, hidden_layer_sizes=(100,), activation="relu", *,
 
     def _validate_input(self, X, y, incremental):
         X, y = self._validate_data(X, y, accept_sparse=['csr', 'csc'],
-                                   multi_output=True)
+                                   multi_output=True,
+                                   dtype=(np.float64, np.float32))
+        self._dtype = X.dtype
```

Comment:
So right now `_validate_input` is indeed only called on fit, but generally I think it would be safer to make it not store anything on `self` in case someone uses it for predict in the future. Jsing `X.dtype` later should be enough no?

Reply:
That makes sense. I had assigned it to `self` as I needed `dtype` in `_init_coef` where `X` is not in scope. Now I am passing `dtype` as a parameter to `_init_coef`.